karapace/schema_registry_apis.py

```diff
@@ -238,8 +223,20 @@ def send_kafka_message(self, key, value):
         future = self.producer.send(self.config["topic_name"], key=key, value=value)
         self.producer.flush(timeout=self.kafka_timeout)
         msg = future.get(self.kafka_timeout)
-        self.log.debug("Sent kafka msg key: %r, value: %r, offset: %r", key, value, msg.offset)
-        self.get_offset_from_queue(msg.offset)
+        sent_offset = msg.offset
+
+        self.log.debug("[Reader Synchronization] Waiting for key: %r, value: %r, offset: %r", key, value, sent_offset)
+        start = time.monotonic()
+
+        while sent_offset > self.ksr.offset:
+            time.sleep(0.1)
```

Comment:
We should use a wait timeout here i think

Reply:
I intentionally left it out, would do in a follow up

Reply:
ah! ok nevermind then! looks good otherwise i think